CMake/BuildFlatBuffers.cmake

```diff
@@ -150,3 +150,321 @@ function(build_flatbuffers flatbuffers_schemas
       ${copy_text_schemas_dir})
   endif()
 endfunction()
+
+# Creates a target that can be linked against that generates flatbuffer headers.
+#
+# This function takes a target name and a list of schemas. You can also specify
+# other flagc flags and options to change the behavior of the flatc compiler.
+# Flags are named the same as they would be in the flatc tool except ALL_CAPS.
+# For example, --gen-mutable would be passed in as GEN_MUTABLE.
+#
+# Example:
+#
+#     flatbuffers_generate_headers(
+#         TARGET generated_headers_target
+#         SCHEMAS ${MY_SCHEMA_FILES}
+#         INCLUDE_PREFIX "${MY_SCHEMA_FILE_DEPENDENCY_DIRECTORY}"
+#         BINARY_SCHEMAS_DIR "${MY_BINARY_SCHEMA_DIRECTORY}"
+#         GEN_OBJECT_API GEN_MUTABLE SCOPED_ENUMS 
+#         REFLECT_TYPES REFLECT_NAMES)
+#
+#     target_link_libraries(MyExecutableTarget
+#         PRIVATE generated_headers_target
+#     )
+function(flatbuffers_generate_headers)
+    set(options 
+    "NO_PREFIX"
+    "SCOPED_ENUMS"
+    "GEN_MUTABLE"
+    "GEN_NAME_STRINGS"
+    "GEN_OBJECT_API"
+    "GEN_COMPARE"
+    "GEN_NULLABLE"
+    "GEN_GENERATED"
+    "GEN_ALL"
+    "CPP_STR_FLEX_CTOR"
+    "OBJECT_PREFIX"
+    "OBJECT_SUFFIX"
+    "RAW_BINARY"
+    "SIZE_PREFIXED"
+    "PROT"
+    "ONEOF_UNION"
+    "GRPC"
+    "BFBS_COMMENT"
+    "KEEP_PREFIX"
+    "REFLECT_TYPES"
+    "REFLECT_NAMES"
+    "FORCE_EMPTY")
+  set(one_value_args
+    "TARGET"
+    "INCLUDE_PREFIX"
+    "BINARY_SCHEMAS_DIR"
+    "ROOT_TYPE"
+    "CPP_PTR_TYPE"
+    "CPP_STR_TYPE"
+    "CONFORM"
+    "CONFORM_INCLUDES"
+    "SOURCE_GROUP_ROOT")
+  set(multi_value_args
+    "SCHEMAS"
+    "INCLUDE"
+    "DEPENDENCIES")
+  cmake_parse_arguments(
+    PARSE_ARGV 0
+    FLATBUFFERS_GENERATE_HEADERS
+    "${options}"
+    "${one_value_args}"
+    "${multi_value_args}")
+
+  macro(option_to_flatc_flag option flatc_flag)
+    if (${FLATBUFFERS_GENERATE_HEADERS_${option}})
+      set(FLATC_ARGS ${FLATC_ARGS} ${flatc_flag})
+    endif()
+  endmacro()
+ 
+  option_to_flatc_flag(NO_PREFIX --no-prefix)
```

Comment:
rather than introducing this whole new set of flags, which will need maintenance as flags in `flatc` change, can we instead have a way to pass `flatc` flags as-is to this function, and simplify this code a lot?

Reply:
I could FLAGS as an option

Reply:
Alright, I made the changes based on your feedback, let me know what you think.

Reply:
Yup, looks a lot better, thanks!